pinot-common/src/main/java/org/apache/pinot/common/utils/ServiceStatus.java

```diff
@@ -236,13 +242,24 @@ public synchronized Status getServiceStatus() {
         return _serviceStatus;
       }
       long now = System.currentTimeMillis();
-      if (now < _endWaitTime) {
-        _statusDescription =
-            String.format("Waiting for consuming segments to catchup, timeRemaining=%dms", _endWaitTime - now);
-        return Status.STARTING;
+      if (now >= _endWaitTime) {
+        _statusDescription = String.format("Consuming segments status GOOD since %dms", _endWaitTime);
+        return Status.GOOD;
       }
-      _statusDescription = String.format("Consuming segments status GOOD since %dms", _endWaitTime);
-      return Status.GOOD;
+      if (_consumptionNotYetCaughtUp && _allConsumingSegmentsHaveReachedLatestOffset.get()) {
+        // TODO: Once the performance of offset based consumption checker is validated:
+        //      - remove the log line
+        //      - uncomment the status & statusDescription lines
+        //      - remove boolean flag _consumptionNotYetCaughtUp
+        _consumptionNotYetCaughtUp = false;
+        LOGGER.info("All consuming segments have reached their latest offsets! "
+            + "Finished {} msec earlier than time threshold.", _endWaitTime - now);
+//      _statusDescription = "Consuming segments status GOOD as all consuming segments have reached the latest offset";
+//      return Status.GOOD;
+      }
+      _statusDescription =
+          String.format("Waiting for consuming segments to catchup, timeRemaining=%dms", _endWaitTime - now);
```

Comment:
suggest adding the value of _consumptionNotYetCaughtUp to the status description. Or, even the number of segments yet to catch up.

Reply:
Added the number of segments yet to catch up.